### PR TITLE
Add Support for Keyboard: TeaQueen

### PR DESCRIPTION
**Pull request by gregandcin**
## Description

This adds support in QMK for the keyboard [TeaQueen](https://github.com/gregandcin/teaqueen). Included is a default keymap and a VIA enabled keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gregandcin/teaqueen/config.h
Comment (zvecr) [inline keyboards/gregandcin/teaqueen/config.h:1]: This file should be removed.

File: keyboards/gregandcin/teaqueen/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    _QWERTY,
+    _SYMB,
+    _NAV,
+    _ADJUST
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE
+};
+
```
Comment (zvecr) [inline keyboards/gregandcin/teaqueen/keymaps/default/keymap.c:20]:
```suggestion
```

File: keyboards/gregandcin/teaqueen/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    _QWERTY,
+    _SYMB,
+    _NAV,
+    _ADJUST
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE
+};
+
```
Comment (zvecr) [inline keyboards/gregandcin/teaqueen/keymaps/via/keymap.c:20]:
```suggestion
```

File: keyboards/gregandcin/teaqueen/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+#define KC_ALAS LALT_T(KC_PAST)
+#define KC_CTPL LCTL_T(KC_BSLS)
+
+#define KC_NAGR LT(_NAV, KC_GRV)
+#define KC_NAMI LT(_NAV, KC_MINS)
+
+#define KC_ADEN LT(_ADJUST, KC_END)
+#define KC_ADPU LT(_ADJUST, KC_PGUP)
+
```
Comment (zvecr) [inline keyboards/gregandcin/teaqueen/keymaps/via/keymap.c:32]:
```suggestion
```

File: keyboards/gregandcin/teaqueen/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+#define KC_ALAS LALT_T(KC_PAST)
+#define KC_CTPL LCTL_T(KC_BSLS)
+
+#define KC_NAGR LT(_NAV, KC_GRV)
+#define KC_NAMI LT(_NAV, KC_MINS)
+
+#define KC_ADEN LT(_ADJUST, KC_END)
+#define KC_ADPU LT(_ADJUST, KC_PGUP)
+
```
Comment (zvecr) [inline keyboards/gregandcin/teaqueen/keymaps/default/keymap.c:32]:
```suggestion
```

File: keyboards/gregandcin/teaqueen/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+		KC_TRNS,          KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS,          KC_TRNS,           KC_TRNS, KC_TRNS,                     KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS),
+
+    [_ADJUST] = LAYOUT(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+		KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS,                     KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS)
```
Comment (zvecr) [inline keyboards/gregandcin/teaqueen/keymaps/via/keymap.c:62]: Dead layer
```suggestion
		KC_TRNS, KC_TRNS,          KC_TRNS,           KC_TRNS, KC_TRNS,                     KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS)
```

File: keyboards/gregandcin/teaqueen/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		KC_TRNS,          KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS,          KC_TRNS,           KC_TRNS, KC_TRNS,                     KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS),
+
+    [_ADJUST] = LAYOUT(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+		KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS,                     KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS)
```
Comment (zvecr) [inline keyboards/gregandcin/teaqueen/keymaps/default/keymap.c:62]: Dead layer
```suggestion
		KC_TRNS, KC_TRNS,          KC_TRNS,           KC_TRNS, KC_TRNS,                     KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS)
```

File: keyboards/gregandcin/teaqueen/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+enum custom_layer {
+    _QWERTY,
+    _SYMB,
+    _NAV,
+    _ADJUST
```
Comment (zvecr) [inline keyboards/gregandcin/teaqueen/keymaps/via/keymap.c:14]:
```suggestion
    _NAV
```

File: keyboards/gregandcin/teaqueen/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+enum custom_layer {
+    _QWERTY,
+    _SYMB,
+    _NAV,
+    _ADJUST
```
Comment (zvecr) [inline keyboards/gregandcin/teaqueen/keymaps/default/keymap.c:14]:
```suggestion
    _NAV
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by gregandcin** [comment]
Removed unused code and files

File: keyboards/gregandcin/teaqueen/info.json [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        "matrix_pins": {
+            "right": {
+                "cols": ["GP29", "GP28", "GP27", "GP26", "GP22", "GP20", "GP23", "GP21"],
+                "rows": ["GP5", "GP6", "GP7", "GP8", "GP9"]
+            }
+        },
+        "soft_serial_pin": "GP3",
+        "transport": {
+            "protocol": "serial"
+        }
```
Comment (waffle87) [inline keyboards/gregandcin/teaqueen/info.json:31]:
```suggestion
        "soft_serial_pin": "GP3"
```

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**